### PR TITLE
Fix: SD29x9 overflow behaviour

### PR DESCRIPTION
**Pull request by immrsd**
- Addressed the #166 issue by catching `SD29x9` overflow arithmetic errors and aborting with a special `EOverflow` code
- `UD30x9` overflow behaviour was already fixed to be consistent in the scope of [PR #199](https://github.com/OpenZeppelin/contracts-sui/pull/199)

## Summary by CodeRabbit

* **New Features**
  * Added absolute value computation for signed fixed-point numbers.

* **Bug Fixes**
  * Improved overflow detection and error handling in arithmetic operations.

* **Documentation**
  * Expanded documentation for fixed-point construction, arithmetic operations, and edge cases across all modules.

* **Tests**
  * Added comprehensive test coverage for boundary conditions and overflow scenarios.

### CODE REVIEW

File: math/fixed_point/sources/sd29x9/sd29x9_base.move [hunk 1]
```diff
@@ -262,9 +269,9 @@ public fun div(x: SD29x9, y: SD29x9): SD29x9 {
     let x = decompose(x.unwrap());
     let y = decompose(y.unwrap());
     let neg = x.neg != y.neg;
-    let numerator = (x.mag as u256) * SCALE_U256;
-    let mag = numerator / (y.mag as u256);
-    wrap_components(Components { neg, mag: mag as u128 })
+    let numerator = x.mag * SCALE;
+    let mag = numerator / y.mag;
+    wrap_components(Components { neg, mag: mag })
```
Comment (0xNeshi) [inline math/fixed_point/sources/sd29x9/sd29x9_base.move]:
```suggestion
    wrap_components(Components { neg, mag })
```
nit

**Review by ericnordelo — APPROVED** [review]
LGTM!
